File: src/main/java/gestorAplicacion/servicios/Tiquete.java

package gestorAplicacion.servicios;

import gestorAplicacion.instalaciones.Instalacion;

import java.io.Serializable;


public class Tiquete  {
    private Instalacion instalacion;
    private float costo;
    private int id;

    public Tiquete(Instalacion instalacion, int id){
        this.instalacion = instalacion;
        this.costo = instalacion.getCosto();
        this.id = id;
    }
    
    

    public void setInstalacion(Instalacion instalacion){
        this.instalacion = instalacion;
    }
    public Instalacion getInstalacion(){
        return instalacion;
    }
    public void setCosto(float costo){
        this.costo = costo;
    }
    public float getCosto(){
        return costo;
    }
    public void setId(int id){
        this.id = id;
    }
    public int getId(){
        return id;
    }
    
    public  boolean comprarTiquete(Tarjeta tj){
        float precio;
        if(tj.getAtracciones()>=5){
             precio = costo*0.8f;
        }
        else{
             precio = costo;
        }
        if(Registro.buscarCliente(tj.idTarjeta).edad >= instalacion.getEdadRestriccion()){
            if(tj.getSaldo() >= precio){
                //if(instalacion.sumarUsoAntes()){
                if(instalacion.sumarUsoAntes()){
                    tj.setSaldo(tj.getSaldo() - precio);
                    tj.sumarAtracciones();
                    return true;
                }
                return false;
            }
        }
        else{
            
            System.out.println("No puede entrar a esta atraccion");
        }
        return false;
    }

    @Override
    public String toString() {
        return "Tiquete{" + "instalacion=" + instalacion + ", costo=" + costo + ", id=" + id + '}';
    }

    
}

File: practica2/src/gestorAplicacion/servicios/tiquete.py

class Tiquete:

    def __init__(self, instalacion, id):
        self._Instalacion = instalacion
        self._costo = instalacion.getCosto()
        self._id = id

    def comprarTiquete(self, tarjeta):
        if tarjeta.getAtracciones() == 5 :
            descuento = int(input("Aplica para desceunto del 20% para su proximo tiquete. Quiere utilizarlo?\n1. Si \n2. No\n"))
            if descuento == 1:
                precio = self._costo*0.8
            else:
                precio = self._costo
        else:
            precio = self._costo

        if self._Instalacion.sumarUsoAntes():
            tarjeta.setSaldo(tarjeta.getSaldo()-precio)
            tarjeta.sumarAtracciones()
            return True
        return print("Esta instalacion se encuentra en mantenimiento")

    def setInstalacion(self, instalacion):
        self._instalacion = instalacion
    
    def getInstalacion(self):
        return self._instalacion
    
    def setCosto(self, costo):
        self._costo = costo
    
    def getCosto(self):
        return self._costo
    
    def setId(self, id):
        self._id = id
    
    def getId(self):
        return self._id
    